src/main/java/mate/academy/internetshop/dao/implementation/BucketDaoImpl.java

```
@@ -28,21 +31,33 @@ public Bucket create(Bucket bucket) {
 
     @Override
     public Bucket update(Bucket bucket) {
-        Bucket buck = Storage.buckets
-                .stream()
-                .filter(x -> x.getId().equals(bucket.getId()))
-                .findFirst()
-                .orElseThrow(NoSuchElementException::new);
-        int index = Storage.buckets.indexOf(buck);
-        return Storage.buckets.set(index, bucket);
+        Bucket buck = null;
+        try {
+            buck = Storage.buckets
+                    .stream()
+                    .filter(x -> x.getId().equals(bucket.getId()))
+                    .findFirst()
+                    .get();
+            int index = Storage.buckets.indexOf(buck);
+            return Storage.buckets.set(index, bucket);
+        } catch (NoSuchElementException e) {
```

@MrNopi Чому ти вирішив відловлювати цей exception і логувати його? Ми говорили на лекції що ексепшини все одно залогуються. В цьому випадку - ти swallow the exception, і це мега погана практика, адже якщо в тебе фейлиться якась операція то щось йде не так. А ігноруючи exception - ти позбавляєш себе можливості виправити проблему через яку в тебе вилетів exception